internal/pkg/scalable/scalableResource.go

```diff
@@ -36,10 +47,10 @@ type Workload interface {
 	GetObjectKind() schema.ObjectKind
 	// SetAnnotations sets the annotations on the resource. Changes won't be made on kubernetes until update() is called
 	SetAnnotations(annotations map[string]string)
-	// SetReplicas sets the amount of replicas on the resource. Changes won't be made on kubernetes until update() is called
-	SetReplicas(replicas int)
-	// GetCurrentReplicas gets the current amount of replicas of the resource
-	GetCurrentReplicas() (int, error)
 	// Update updates the resource with all changes made to it. It should only be called once on a resource
-	Update(clientset *kubernetes.Clientset, ctx context.Context) error
+	Update(clientset *kubernetes.Clientset, dynamicClient dynamic.Interface, ctx context.Context) error
+	// ScaleUp upscale the resource when the downscale period ends
```

Comment:
the comments for ScaleUp and ScaleDown should also be updated in the workload interface